Source/Client/Multiplayer.cs

```diff
@@ -193,16 +193,19 @@ private static void HandleCommandLine()
         {
             if (GenCommandLine.TryGetCommandLineArg("connect", out string ip))
             {
-                int port = MultiplayerServer.DefaultPort;
+                int port;
 
-                var split = ip.Split(':');
-                if (split.Length == 0)
+                MatchCollection matches = Regex.Matches(ip, "(.*)((?::))((?:[0-9]+))$");
+                if (matches.Count == 1) // Port is included, use regex
+                {
+                    ip = matches[0].Groups[1].Value;
```

Comment:
If `ip` contained an IPv6 with port, it would have been of the form [`[some:ipv6::address]:port`](https://en.wikipedia.org/wiki/IPv6_address#Literal_IPv6_addresses_in_network_resource_identifiers), so a special case would need to be created for trimming the brackets. I'd suggest using a different regexp for this case, or even doing some plain string checks.
I might take a look on this after work today.